// react-d3-tooltip/lib/utils/tooltip.js
"use strict";
// Please check bottom propTypes to know which props is accepted.

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _table = require('../tooltip/table');

var _table2 = _interopRequireDefault(_table);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Tooltip = function (_Component) {
  _inherits(Tooltip, _Component);

  function Tooltip(props) {
    _classCallCheck(this, Tooltip);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(Tooltip).call(this, props));
  }

  _createClass(Tooltip, [{
    key: 'render',
    value: function render() {
      var _props = this.props;
      var xTooltip = _props.xTooltip;
      var yTooltip = _props.yTooltip;
      var contentTooltip = _props.contentTooltip;
      var dist = _props.dist;


      var contentTooltipTmpl = undefined;

      var style = {
        left: xTooltip ? xTooltip + dist : -10000,
        top: yTooltip ? yTooltip + dist : -10000,
        position: 'fixed'
      };

      if (contentTooltip) {
        if (this.props.children) {
          contentTooltipTmpl = _react2.default.cloneElement(this.props.children, { contentTooltip: contentTooltip });
        } else {
          contentTooltipTmpl = _react2.default.createElement(_table2.default, { contentTooltip: contentTooltip });
        }
      }

      return _react2.default.createElement(
        'div',
        {
          style: style,
          className: 'react-d3-basics__tooltip_utils'
        },
        contentTooltipTmpl
      );
    }
  }]);

  return Tooltip;
}(_react.Component);

Tooltip.defaultProps = {
  gravity: 's',
  dist: 15
};
exports.default = Tooltip;
module.exports = exports['default'];
