// react-d3-tooltip/lib/tooltip/table/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var SimpleTooltipStyle = function (_Component) {
  _inherits(SimpleTooltipStyle, _Component);

  function SimpleTooltipStyle(props) {
    _classCallCheck(this, SimpleTooltipStyle);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(SimpleTooltipStyle).call(this, props));
  }

  _createClass(SimpleTooltipStyle, [{
    key: 'render',
    value: function render() {
      var contentTooltip = this.props.contentTooltip;

      var cv = Object.keys(contentTooltip).map(function (d, i) {
        if (d === 'color') {
          var colorStyle = {
            backgroundColor: contentTooltip[d],
            color: '#FFF'
          };
        }

        var trStyle = {
          display: 'table-row',
          backgroundImage: 'linear-gradient(#FFF, #EEE)',
          padding: '3px',
          height: '30px'
        };

        var tdStyle = {
          display: 'table-cell',
          padding: '3px',
          verticalAlign: 'middle',
          whiteSpace: 'normal',
          border: '1px solid #D3D3D3'
        };

        var tdHeadStyle = {
          display: 'table-cell',
          padding: '3px',
          verticalAlign: 'middle',
          whiteSpace: 'normal',
          border: '1px solid #D3D3D3',
          backgroundColor: '#555',
          color: '#FFF',
          textTransform: 'capitalize'
        };

        if (colorStyle) {
          var tdColorStyle = Object.assign(tdStyle, colorStyle);
        }

        return _react2.default.createElement(
          'div',
          { className: 'tooltip_tr', style: trStyle, key: i },
          _react2.default.createElement(
            'div',
            { className: 'tooltip_td', style: tdHeadStyle, key: i },
            d
          ),
          _react2.default.createElement(
            'div',
            { className: 'tooltip_td', style: colorStyle ? tdColorStyle : tdStyle, key: i.i },
            contentTooltip[d]
          )
        );
      });

      return _react2.default.createElement(
        'div',
        null,
        cv
      );
    }
  }]);

  return SimpleTooltipStyle;
}(_react.Component);

exports.default = SimpleTooltipStyle;
module.exports = exports['default'];
