// react-d3-tooltip/lib/inherit/barEvt.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var BarEvt = function (_Component) {
  _inherits(BarEvt, _Component);

  function BarEvt(props) {
    _classCallCheck(this, BarEvt);

    var _this = _possibleConstructorReturn(this, Object.getPrototypeOf(BarEvt).call(this, props));

    _this.state = {
      xTooltip: null,
      yTooltip: null,
      contentTooltip: null
    };
    return _this;
  }

  _createClass(BarEvt, [{
    key: "mouseOut",
    value: function mouseOut(e, d) {

      this.setState({
        xTooltip: null,
        yTooltip: null,
        contentTooltip: null
      });
    }
  }, {
    key: "mouseOver",
    value: function mouseOver(e, d) {

      var contentTooltip = { title: d.name, value: d.y, fieldTitle: d.x, color: d.color };
      this.setState({
        xTooltip: e.clientX,
        yTooltip: e.clientY,
        contentTooltip: contentTooltip
      });
    }
  }]);

  return BarEvt;
}(_react.Component);

exports.default = BarEvt;
module.exports = exports['default'];
