// react-d3-tooltip/lib/inherit/voronoiEvt.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var VoronoiEvt = function (_Component) {
  _inherits(VoronoiEvt, _Component);

  function VoronoiEvt(props) {
    _classCallCheck(this, VoronoiEvt);

    var _this = _possibleConstructorReturn(this, Object.getPrototypeOf(VoronoiEvt).call(this, props));

    _this.state = {
      focusX: -10000,
      focusY: -10000,
      xTooltip: null,
      yTooltip: null,
      contentTooltip: null
    };
    return _this;
  }

  _createClass(VoronoiEvt, [{
    key: "voronoiMouseOut",
    value: function voronoiMouseOut(e, d) {
      this.setState({
        focusX: -10000,
        focusY: -10000,
        xTooltip: null,
        yTooltip: null,
        contentTooltip: null
      });
    }
  }, {
    key: "voronoiMouseOver",
    value: function voronoiMouseOver(e, d, xScaleSet, yScaleSet, stack) {
      var newY = stack ? yScaleSet(d.y + d.y0) : yScaleSet(d.y);
      var contentTooltip = { title: d.x.toString(), fieldTitle: d.name.toString(), value: d.y.toString(), color: d.color };
      window.alert("in voronoi");
      this.setState({
        focusX: xScaleSet(d.x),
        focusY: newY,
        xTooltip: e.clientX,
        yTooltip: e.clientY,
        contentTooltip: contentTooltip
      });
    }
  }]);

  return VoronoiEvt;
}(_react.Component);

exports.default = VoronoiEvt;
module.exports = exports['default'];
